src/main/java/uk/gov/hmcts/reform/coh/service/QuestionService.java

```diff
@@ -111,4 +125,52 @@ public Question updateQuestionForced(Question question){
     public void deleteQuestion(Question question) {
         questionRepository.delete(question);
     }
+
+    @Transactional
+    public void requestDeadlineExtension(OnlineHearing onlineHearing) {
+        List<Question> questions = findAllQuestionsByOnlineHearing(onlineHearing)
+            .orElseThrow(() -> new RuntimeException("Could not retrieve questions"));
+
+        if (questions.isEmpty()) {
+            throw new RuntimeException("There are no questions to be answered");
+        }
+
+        Instant now = Instant.now();
+
+        questions.stream()
+            .filter(question -> now.isBefore(question.getDeadlineExpiryDate().toInstant()))
+            .forEach(question -> {
+                if (shouldGrantExtensionTo(question)) {
+                    grantDeadlineExtension(question);
+                } else if (shouldDenyExtensionTo(question)) {
+                    denyDeadlineExtension(question);
+                }
+            });
```

Comment:
Would be simpler just to have 
`                if (shouldGrantExtensionTo(question)) {
                    grantDeadlineExtension(question);
                } else {
                    denyDeadlineExtension(question);
                }
`
And shouldGrantExtensionTo would just have all the logic, no?

Reply:
There are 3 types of questions:
1. the ones that can be granted extension
2. the ones that can be denied extension
3. the ones that should be ignored by the decision, i.e. pending issue, already expired

Now, the 3rd type would be treated as the 2nd type, which is not recommended.

Reply:
Ah yes. Fair enough